R/join_key.R

```diff
@@ -0,0 +1,101 @@
+#' Create a relationship between a pair of datasets
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' @details `join_key()` will create a relationship for the variables on a pair
+#' of datasets.
+#'
+#' @inheritParams join_keys
+#' @param dataset_2 (optional `character`) other dataset name. In case it is omitted, then it
+#' will create a primary key for `dataset_1`.
+#' @param keys (optionally named `character`) where `names(keys)` are columns in `dataset_1`
+#' with relationship to columns of `dataset_2` given by the elements in `keys`.
+#' If `names(keys)` is `NULL` then the same column names are used for both `dataset_1`
+#' and `dataset_2`.
+#'
+#' @return object of class `join_key_set` to be passed into `join_keys` function.
+#'
+#' @seealso [join_keys()]
+#'
+#' @export
+#'
+#' @examples
+#' join_key("d1", "d2", c("A"))
+#' join_key("d1", "d2", c("A" = "B"))
+#' join_key("d1", "d2", c("A" = "B", "C"))
+#' join_key("d1", "d2", c("A" = "B", "C" = ""))
+join_key <- function(dataset_1, dataset_2 = dataset_1, keys) {
+  checkmate::assert_string(dataset_1)
+  checkmate::assert_string(dataset_2)
+  checkmate::assert_character(keys, any.missing = FALSE)
+
+  if (length(keys) > 0) {
```

Comment:
Why is `character(0)` allowed? `NULL` is forbidden by assertion in line 30.

Reply:
I'm not sure. What does this reflect? That would be a no-relation, right? Unless I'm mistaken we don't normally keep them in the objects. When a join_key is passed to `*_join`, reaching for a non-existent list element returns `NULL` and that is accepted in the `by` argument`.

Reply:
Then let's scrap `character(0)` altogether :grin: